keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -131,75 +129,83 @@ Sidecar start up will follow these steps:
-1. The sidecar detects this and issues a _DeleteBucket()_ rpc to the driver.  It passes pertinent data stored in the `BucketContent`.
-1. The driver deletes the bucket from the object store and responds with no error.
-1. The sidecar sets the `BucketContent`'s status to indicate the delete occurred.
-1. The COSI controller deletes the `BucketContent` and app secret.
-1. The COSI controller removes the `Bucket`, app secret, and `BucketContent` finalizers so they are garbage collected.
 
-### Brownfield Buckets
 
-#### Grant Access
-
-1. A `BucketClass` is defined specifically referencing an object store bucket.
-1. A user creates a `Bucket` in the app namespace, specifying the `BucketClass`.
-1. The COSI controller detects the `Bucket` and creates a `BucketContent` object.
-1. The sidecar detects the `BucketContent` object and calls the _GrantAccess()_ rpc to the driver, returing a set of credentials for accessing the bucket.
-1. The sidecar writes the credentials and endpoint information to a Secret in its namespace, with an owner reference to the BucketContent.
-1. The COSI controller copies the sidecar's secret to the `Bucket`'s namespace.
-1. The COSI controller adds a finalizer to the `BucketContent`, `Bucket`, and app secret.
-1. The COSI controller sets owner references on the `Bucket` and secret. Note: the `BucketContent` being cluster-scoped does not have an owner reference.
-1. The workflow ingests the Secret to begin operation.
-
-#### Revoke Access
-
-1. The user deletes the `Bucket`, which blocks until revoke operations complete.
-1. The COSI controller detects the update and deletes the `BucketContent`, which also blocks until backend revoke operations complete.
-1. The sidecar detects the `BucketContent` update and invokes the _RevokeAccess()_ rpc method.
-1. The driver removes access for the associated credentials.
-1. The sidecar removes the finalizer from and deletes its Secret.
-1. The sidecar sets the `BucketContent`'s status to indicate the revoke occurred.
-1. The COSI controller deletes the `BucketContent` and app secret.
-1. The COSI controller removes the `Bucket`, app secret, and `BucketContent` finalizers so they are garbage collected.
+### Distinguishing Bucket Case
+
+| BucketClassFields             | SecretRef: nil | SecretRef: non-nil |
+| ----------------------------- | -------------- | ------------------ |
+| **objectBucketName: non-nil** | Brownfield     | Undefined          |
+| **objectBucketName: nil**     | Greenfield     | Static             |
+
+#### Create Bucket (Greenfield)
+
+1. The user creates a `Bucket` in their namespace, with reference to a `BucketClass`.
+1. The Controller sees the new `Bucket` and applies a `finalizer` for orchestrated deletions.
+1. The Controller gets the `BucketClass` referenced by the `Bucket`.
+1. The Controller creates a `BucketContent` object with its `BucketClassName` set to the name of its `BucketClass` and a `finalizer`.
```

Comment:
nit: How about "...a `BucketContent` object with its `BucketClassName` set to the name of **the** `BucketClass`..."